src/main/java/net/revelc/code/formatter/xml/eclipse/XmlDocumentFormatter.java

```diff
@@ -0,0 +1,441 @@
+/*******************************************************************************
+ * Copyright (c) 2004, 2011 John-Mason P. Shackelford and others.,
+ *               2009 Jose Montoya
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ *
+ * Contributors:
+ *     John-Mason P. Shackelford - initial API and implementation
+ * 	   IBM Corporation - bug fixes
+ * 	   Jose Montoya - Modified implementation outside Eclipse Platform
+ *******************************************************************************/
+package net.revelc.code.formatter.xml.eclipse;
+
+import java.io.IOException;
+import java.io.Reader;
+import java.io.StringReader;
+
+/**
+ * Based on Eclipse Ant Formatter and distributed under original Eclipse 2.0 license and compatible Apache 2.0 See:
+ * https://github.com/eclipse/eclipse.platform/tree/master/ant/org.eclipse.ant.ui/Ant%20Editor/org/eclipse/ant/internal/ui/editor/formatter
```

Comment:
If this code is already available in a separate library, can that library simply be pushed to Maven Central and reused? I'm not a fan of copying external code in when a dependency will do.